scipy/stats/tests/test_resampling.py

```diff
@@ -732,124 +737,197 @@ def statistic_extradim(*args, axis):
-            monte_carlo_test([1, 2, 3], stats.norm.rvs, stat, vectorized=1.5)
+            monte_carlo_test(data, stats.norm.rvs, stat, vectorized=1.5)
 
         message = "`rvs` must be callable or sequence of callables."
         with pytest.raises(TypeError, match=message):
-            monte_carlo_test([1, 2, 3], None, stat)
+            monte_carlo_test(data, None, stat)
         with pytest.raises(TypeError, match=message):
-            monte_carlo_test([[1, 2], [3, 4]], [lambda x: x, None], stat)
+            temp = xp.asarray([[1., 2.], [3., 4.]])
+            monte_carlo_test(temp, [lambda x: x, None], stat)
 
         message = "If `rvs` is a sequence..."
         with pytest.raises(ValueError, match=message):
-            monte_carlo_test([[1, 2, 3]], [lambda x: x, lambda x: x], stat)
+            temp = xp.asarray([[1., 2., 3.]])
+            monte_carlo_test(temp, [lambda x: x, lambda x: x], stat)
 
         message = "`statistic` must be callable."
         with pytest.raises(TypeError, match=message):
-            monte_carlo_test([1, 2, 3], stats.norm.rvs, None)
+            monte_carlo_test(data, stats.norm.rvs, None)
 
         message = "`n_resamples` must be a positive integer."
         with pytest.raises(ValueError, match=message):
-            monte_carlo_test([1, 2, 3], stats.norm.rvs, stat,
-                             n_resamples=-1000)
+            monte_carlo_test(data, stats.norm.rvs, stat, n_resamples=-1000)
 
         message = "`n_resamples` must be a positive integer."
         with pytest.raises(ValueError, match=message):
-            monte_carlo_test([1, 2, 3], stats.norm.rvs, stat,
-                             n_resamples=1000.5)
+            monte_carlo_test(data, stats.norm.rvs, stat, n_resamples=1000.5)
 
         message = "`batch` must be a positive integer or None."
         with pytest.raises(ValueError, match=message):
-            monte_carlo_test([1, 2, 3], stats.norm.rvs, stat, batch=-1000)
+            monte_carlo_test(data, stats.norm.rvs, stat, batch=-1000)
 
         message = "`batch` must be a positive integer or None."
         with pytest.raises(ValueError, match=message):
-            monte_carlo_test([1, 2, 3], stats.norm.rvs, stat, batch=1000.5)
+            monte_carlo_test(data, stats.norm.rvs, stat, batch=1000.5)
 
         message = "`alternative` must be in..."
         with pytest.raises(ValueError, match=message):
-            monte_carlo_test([1, 2, 3], stats.norm.rvs, stat,
-                             alternative='ekki')
+            monte_carlo_test(data, stats.norm.rvs, stat, alternative='ekki')
+
+        # *If* this raises a value error, make sure it has the intended message
+        message = "Signature inspection of statistic"
+        def rvs(size):
+            return xp.asarray(stats.norm.rvs(size=size))
+        try:
+            monte_carlo_test(data, rvs, xp.mean)
+        except ValueError as e:
+            assert str(e).startswith(message)
```

Comment:
Can this not be done with the standard testing infrastructure?

Reply:
I'm not sure I follow what the link is adding. `raises` can check for partial matches, sure, but it requires that the exception be raised. This exception only occurs when introspection is not possible (e.g. PyTorch). The comments notes:

> *If* this raises a value error, make sure it has the intended message